src/SFA.DAS.RoATPService.Application/Handlers/UpdateOrganisationProviderTypeHandler.cs

```diff
@@ -1,92 +1,179 @@
+        }
 
-            bool success = await _updateOrganisationRepository.UpdateProviderType(request.OrganisationId, request.ProviderTypeId,
-                request.OrganisationTypeId, request.UpdatedBy);
+        private async Task<bool> ProcessOrganisationsDetailsAndUpdateAuditStatusAndStartDate(Guid organisationId, string updatedBy, int providerTypeId, int previousProviderTypeId, 
+            int previousOrganisationStatusId, DateTime? previousStartDate, AuditData auditData)
+        {
+            var changeStatusToActiveAndSetStartDate = 
+                _organisationStatusManager.ShouldChangeStatustoActiveAndSetStartDateToToday(providerTypeId, previousProviderTypeId, previousOrganisationStatusId);
 
-            if (success)
+          
+            if (changeStatusToActiveAndSetStartDate)
             {
-                var auditData = CreateAuditData(request.OrganisationId, request.UpdatedBy);
+                if (previousOrganisationStatusId != OrganisationStatus.Active)
+                {
+                    var success = await _updateOrganisationRepository.UpdateStatus(organisationId,
+                        OrganisationStatus.Active, updatedBy);
 
-                AddAuditEntry(
-                    auditData, 
-                    "Provider Type", 
-                    GetProviderType(previousProviderTypeId).Result, 
-                    GetProviderType(request.ProviderTypeId).Result
-                    );
+                    if (!success)
+                    {
+                        return false;
+                    }
 
-                if (previousOrganisationTypeId != request.OrganisationTypeId)
+                    AddAuditEntry(
+                        auditData,
+                        "Organisation Status",
+                        GetOrganisationStatus(previousOrganisationStatusId).Result,
+                        GetOrganisationStatus(OrganisationStatus.Active).Result
+                    );
+                }
+           
+                if (previousStartDate == null || previousStartDate.Value.Date != DateTime.Today.Date)
                 {
+                    var success = await _updateOrganisationRepository.UpdateStartDate(organisationId, DateTime.Today);
+
+                    if (!success) { return false;}
+
                     AddAuditEntry(
                         auditData,
-                        "Organisation Type",
-                        GetOrganisationType(previousOrganisationTypeId, previousProviderTypeId).Result,
-                        GetOrganisationType(request.OrganisationTypeId, request.ProviderTypeId).Result
+                        "Start Date",
+                        previousStartDate?.ToString(),
+                        DateTime.Today.ToString(CultureInfo.InvariantCulture)
                     );
                 }
-                
-                return await _auditLogRepository.WriteFieldChangesToAuditLog(auditData);
             }
```

Comment:
Should it drop out of the process here  if changeStatusToActiveAndSetStartDate = true, as an org won't qualify both for being active and on-boarding?